Doc changes for pausing MHC before an upgrade (ECOPROJECT-143) (#36102)
openshift/openshift-docs: modules/machine-health-checks-pausing.adoc

// Module included in the following assemblies:

// * updating/updating-cluster-cli.adoc

[id="machine-health-checks-pausing_{context}"]
= Pausing a MachineHealthCheck resource

> During the upgrade process, nodes in the cluster might become temporarily unavailable. The `MachineHealthCheck` might identify such nodes as unhealthy and reboot them. To avoid rebooting such nodes, pause all the `MachineHealthCheck` resources before updating the cluster.

Reviewer: The styling for specific API objects and generic classes of api objects are different: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

I think you mean this:

```suggestion
During the upgrade process, nodes in the cluster might become temporarily unavailable. The machine health check might identify such nodes as unhealthy and reboot them. To avoid rebooting such nodes, pause all the `MachineHealthCheck` resources before updating the cluster.
```

.Prerequisites

> * Install the `oc` command line interface.

Reviewer: There's a standard way to say this: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#openshift-cli

```suggestion
* Install the OpenShift CLI (`oc`).
```

.Procedure

> * To pause the machine health checks, add `cluster.x-k8s.io/paused=""` annotation to the `MachineHealthCheck` resource. To do so, run the following command:

Reviewer: I think you can tighten this a little:

```suggestion
* To pause the machine health checks, add the `cluster.x-k8s.io/paused=""` annotation to the `MachineHealthCheck` resource. Run the following command:
```

+
[source,terminal]
----

> $ oc -n <namespace> annotate mhc <mhc-name> cluster.x-k8s.io/paused=""

Reviewer: Is the user going to have the information for these variable on hand? At some point, it might be a good improvement to point out the default namespace name, `openshift-machine-api`, and how to get the mhc resource name.

Reviewer: +1 to upgrade, not installation.

Author: I have now updated the commands and added a step to get the MHC resource name.

----

> 

Reviewer: I think the rest of the procedure should nest under the command.

```suggestion
+
```

The annotated machine health check resource resembles the following YAML file:

> 

Reviewer: ```suggestion
+
```

[source,yaml]
----
apiVersion: machine.openshift.io/v1beta1
kind: MachineHealthCheck
metadata:
  name: example
  namespace: openshift-machine-api
  annotations:
    cluster.x-k8s.io/paused: ""
spec:
  selector:
    matchLabels:
      role: worker
  unhealthyConditions:
  - type:    "Ready"
    status:  "Unknown"
    timeout: "300s"
  - type:    "Ready"
    status:  "False"
    timeout: "300s"
  maxUnhealthy: "40%"
status:
  currentHealthy: 5
  expectedMachines: 5
----

> 

Reviewer: ```suggestion
+
```

[IMPORTANT]
====

> Resume the machine health checks after updating the cluster. To resume the check, remove the pause annotation from the `MachineHealthCheck` resource as follows:

Reviewer: This is purely a suggestion because I think it's clearer to be explicit about running commands:

```suggestion
Resume the machine health checks after updating the cluster. To resume the check, remove the pause annotation from the `MachineHealthCheck` resource by running the following command:
```

// Module included in the following assemblies:

// * updating/updating-cluster-cli.adoc
// * updating/updating-cluster-between-minor.adoc
// * updating/updating-restricted-network-cluster.adoc

[id="machine-health-checks-pausing_{context}"]
= Pausing a MachineHealthCheck resource

> During the upgrade process, nodes in the cluster might become temporarily unavailable. In case of worker nodes, the machine health check might identify such nodes as unhealthy and reboot them. To avoid rebooting such nodes, pause all the `MachineHealthCheck` resources before updating the cluster.

Reviewer 2: s/In case of worker nodes/In the case of worker nodes

.Prerequisites

* Install the OpenShift CLI (`oc`).

.Procedure

> 1. To list all the available `MachineHealthCheck` resources that you want to pause, run the following command:

Reviewer 2: These enumerations shouldn't be hardcoded. I would suggest using `. ` for an ordered list here and on line 25.

+
[source,terminal]
----
$ oc get machinehealthcheck -n openshift-machine-api
----

2. To pause the machine health checks, add the `cluster.x-k8s.io/paused=""` annotation to the `MachineHealthCheck` resource. Run the following command:
+
[source,terminal]
----
$ oc -n openshift-machine-api annotate mhc <mhc-name> cluster.x-k8s.io/paused=""
----
+

> The annotated machine health check resource resembles the following YAML file:

Reviewer 2: I'd recommend either using the [general](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references) reference as "machine health check" (without "resource") or the [actual value](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references) as "`MachineHealthCheck` resource" to align with our guidelines.
